plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java

```diff
@@ -48,6 +52,30 @@ public DeployImageWizard(IDeployImageParameters wizardModel) {
 
 	@Override
 	public void addPages() {
+	    if (getModel().originatedFromDockerExplorer()) {
+	        addPage(new ConnectionWizardPage(this, new IConnectionAware<IConnection>() {
```

Comment:
In this snippet cast gives a compilation error:

	void test1() {
		class A {};
		class B extends A {};
		class C<A> {};
		C<A> c = new C<A>();
		C<B> cb = new C<B>();
		c = (C<B>)cb;
	}

This one is correct:

	void test2() {
		class A {};
		class B extends A {};
		class C<A> {};
		C<? extends A> c = new C<A>();
		C<B> cb = new C<B>();
		c = (C<B>)cb;
	}

Sometimes it is very annoying, for example, if I want to store and operate a List<MyImpl> but have to return List<MyInterface> (MyImpl implements MyInterface), a cannot return the list as is, it does not cast.

Still, there is a sense in it. ConnectionWizardPage does not mind accepting any implementation of IConnection, but it should be unambiguously stated with <? extends IConnection> (or with a type parameter).